extension/parquet/column_writer.cpp

```diff
@@ -1152,6 +1152,19 @@ class StringStatisticsState : public ColumnWriterStatistics {
 	}
 };
 
+struct StringHash {
+	std::size_t operator()(const string_t &k) const {
+		return Hash(k);
+	}
+};
+
+struct StringEquality {
+	bool operator()(const string_t &a, const string_t &b) const {
+		return Equals::Operation(a, b);
+	}
+};
+typedef unordered_map<string_t, uint32_t, StringHash, StringEquality> string_dictionary_t;
```

Comment:
Could we move this to a separate header and unify it with the equivalent string dictionary in `dictionary_compression.cpp`?